rules/test/lldb/lldb_sim_runner.py

```diff
@@ -174,26 +171,11 @@ class TestContext():
     def __init__(self, pid, app_name, test_root):
         self.app_name = app_name
         self.test_root = test_root
-
         self.status = None
         self.pid = None
         self.udid = None
-
-        # Bazel sends the MacOS SDK for py interpreter
-        if "SDKROOT" in os.environ:
-            self.developer_dir = os.path.dirname(os.path.dirname(
-                os.path.dirname(os.path.dirname(os.path.dirname(os.environ["SDKROOT"])))))
-        else:
-            xcode_select_result = subprocess.run(["xcode-select", "-p"],
-                                                 encoding="utf-8",
-                                                 check=True,
-                                                 stdout=subprocess.PIPE)
-
-            self.developer_dir = xcode_select_result.stdout.rstrip()
-        os.environ["DEVELOPER_DIR"] = self.developer_dir
-        self.simctl_path = os.path.join(
-            self.developer_dir, "usr", "bin", "simctl")
```

Comment:
This setting of the environ was causing issues in testing, seemed to be changing the xcode selected path. Updated this to use `xcrun -f` to find the path instead and it seems to work more stably